src/test/java/com/adobe/target/edge/client/http/DefaultTargetHttpClientTest.java

```diff
@@ -93,6 +100,20 @@ void testProxyConfigSetWithAuthentication() {
     targetClient.close();
   }
 
+  @Test
+  void testConfigSetWithSSLFactory() throws Exception {
+    SSLContext context = SSLContextBuilder.create().build();
+    SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(context);
+    CloseableHttpClient httpClient =
```

Comment:
I think the interface that CloseableHttpClient implements ([HttpClient](https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/HttpClient.html)) is the client that we actually want to expose inside ClientConfig.  And then in the SDK we can call `new ApacheClient(httpClient, httpClientConfig)`.  That way the customer does not need to worry about the kong.unirest http implementation - which is really an internal SDK implementation detail.